tools/thinpool.sh

```diff
@@ -24,22 +24,51 @@ if [ -z "$name" ]; then
     exit 0
 fi
 
-dm_device="/dev/mapper/$(echo ${VOLUME_GROUP} | sed -e s/-/--/g)-$name"
+if [[ -z "$FICD_DM_VOLUME_GROUP" ]]; then
+    pool_create() {
+        echo
+    }
+
+    pool_remove() {
+        echo
+    }
+
+    pool_reset() {
+        local dev_no=1
+        while true; do
+            sudo dmsetup message "$name" 0 "delete $dev_no" || break
+            dev_no=$(($dev_no + 1))
+        done
+    }
+else
+    dm_device="/dev/mapper/$(echo ${FICD_DM_VOLUME_GROUP} | sed -e s/-/--/g)-$name"
 
-pool_create() {
-    sudo lvcreate --type thin-pool \
-         --poolmetadatasize 16GiB \
-         --size 1G \
-         -n "$name" "$VOLUME_GROUP"
-}
+    pool_create() {
+        echo sudo lvcreate --type thin-pool \
+             --poolmetadatasize 16GiB \
+             --size 1G \
+             -n "$name" "$FICD_DM_VOLUME_GROUP"
+        sudo lvcreate --type thin-pool \
+             --poolmetadatasize 16GiB \
+             --size 1G \
+             -n "$name" "$FICD_DM_VOLUME_GROUP"
+    }
 
-pool_remove() {
-    sudo dmsetup remove "${dm_device}-snap-"* || true
-    sudo dmsetup remove \
+    pool_remove() {
+        sudo dmsetup remove "${dm_device}-snap-"* || true
```

Comment:
As far as I know, `dmsetup remove` does not support masks, so this will always fail.

Reply:
Sorry, which masks?

Reply:
`"${dm_device}-snap-"*`
As far as I understand this meant to be `dmsetup remove devicename-snap-*` ?

Reply:
Yeah, but `*` will be expanded by a shell. So dmsetup won't see `*`